### PR TITLE
Implement limited X509_REQ functionality

### PR DESCRIPTION
**Pull request by jethrogb**
This PR adds some functionality dealing with certificate signing requests. In particular:
- Reading from PEM format
- Writing to PEM format
- Creating them with X509Generator

### CODE REVIEW

File: openssl/src/x509/mod.rs [hunk 1]
```diff
@@ -538,6 +552,50 @@ impl <'x> X509Name<'x> {
     }
 }
 
+#[allow(dead_code)]
```
Comment (sfackler) [inline openssl/src/x509/mod.rs]: This dead code annotation should be able to be removed.

**Comment by sfackler** [comment]
Looks good other than one small nit, thanks!

File: openssl/src/x509/mod.rs (hunk 1 above)
Comment (jethrogb) [inline openssl/src/x509/mod.rs]: Oops, that got copied from `struct X509`.

**Comment by sfackler** [comment]
Thanks!

**Pull request merged**